src/lib/types/src/Flow/Types/Type/Logical/HTMLType.php

```diff
@@ -24,24 +34,38 @@ public function assert(mixed $value) : HTMLDocument
 
     public function cast(mixed $value) : HTMLDocument
     {
-        if ($this->isValid($value)) {
-            return $value;
-        }
-
-        if (!is_string($value) && !is_object($value)) {
+        if (!$this->isValid($value)) {
             throw new CastingException($value, $this);
         }
 
-        try {
-            return new HTMLDocument($value);
-        } catch (InvalidArgumentException $e) {
-            throw new CastingException($value, $this, $e);
+        /* @phpstan-ignore-next-line */
+        if (\is_string($value)) {
+            return HTMLDocument::createFromString($value);
         }
+
+        return $value;
     }
 
     public function isValid(mixed $value) : bool
     {
-        return $value instanceof HTMLDocument;
+        // \Dom\HTMLDocument exist in PHP 8.4+
+        if (\PHP_VERSION_ID < 80400) {
+            return false;
+        }
+
+        if (!\is_string($value)) {
+            return $value instanceof HTMLDocument;
+        }
+
+        if ('' === $value) {
+            return false;
+        }
+
+        if ('<' !== $value[0]) {
+            return false;
+        }
+
+        return \preg_match(self::HTML_ALIKE_REGEX, $value) === 1;
```

Comment:
all of the above is incorrect, TypeHTML expects `@implements Type<HTMLDocument>` - this means that string, even looking like html is not a valid HTMLDocument